homeassistant/components/homekit/thermostats.py

```diff
@@ -0,0 +1,137 @@
+"""Class to hold all thermostat accessories."""
+import logging
+from threading import Timer
+
+from homeassistant.components.climate import (
+    ATTR_CURRENT_TEMPERATURE, ATTR_TEMPERATURE,
+    ATTR_OPERATION_MODE, STATE_HEAT, STATE_COOL)
+from homeassistant.const import ATTR_ENTITY_ID
+from homeassistant.helpers.event import async_track_state_change
+
+from . import TYPES
+from .accessories import HomeAccessory, add_preload_service
+from .const import (
+    SERV_THERMOSTAT, CHAR_CURRENT_HEATING_COOLING,
+    CHAR_TARGET_HEATING_COOLING, CHAR_CURRENT_TEMPERATURE,
+    CHAR_TARGET_TEMPERATURE, CHAR_TEMP_DISPLAY_UNITS)
+
+_LOGGER = logging.getLogger(__name__)
+
+UNIT_HASS_TO_HOMEKIT = {'°C': 0, '°F': 1}
+UNIT_HOMEKIT_TO_HASS = {c: s for s, c in UNIT_HASS_TO_HOMEKIT.items()}
+HC_HASS_TO_HOMEKIT = {'off': 0, STATE_HEAT: 1, STATE_COOL: 2}
+HC_HOMEKIT_TO_HASS = {c: s for s, c in HC_HASS_TO_HOMEKIT.items()}
+
+
+@TYPES.register('Thermostat')
+class Thermostat(HomeAccessory):
+    """Generate a Thermostat accessory for an alarm control panel."""
+
+    def __init__(self, hass, entity_id, display_name):
+        """Initialize a Thermostat accessory object."""
+        super().__init__(display_name, entity_id, 'THERMOSTAT')
+
+        self._hass = hass
+        self._entity_id = entity_id
+        self._call_timer = None
+
+        self.heat_cool_flag_target_state = False
+        self.temperature_flag_target_state = False
+
+        self.service_thermostat = add_preload_service(self, SERV_THERMOSTAT)
+        self.char_current_heat_cool = self.service_thermostat. \
+            get_characteristic(CHAR_CURRENT_HEATING_COOLING)
+        self.char_current_heat_cool.value = 0
+        self.char_target_heat_cool = self.service_thermostat. \
+            get_characteristic(CHAR_TARGET_HEATING_COOLING)
+        self.char_target_heat_cool.value = 0
+        self.char_target_heat_cool.setter_callback = self.set_heat_cool
+
+        self.char_current_temp = self.service_thermostat. \
+            get_characteristic(CHAR_CURRENT_TEMPERATURE)
+        self.char_current_temp.value = 0.0
+        self.char_target_temp = self.service_thermostat. \
+            get_characteristic(CHAR_TARGET_TEMPERATURE)
+        self.char_target_temp.value = 0.0
+        self.char_target_temp.setter_callback = self.set_temperature_debounced
```

Comment:
Use `self.set_target_temperature` instead. We shouldn't use `self.set_temperature_debounced` as I've explained.

Reply:
I removed the debounced version based on your request, even though I still think it is needed. When you use the standard iOS Home app, you have to use a slider to set the temperature, resulting in a lot of unnecessary calls.

Reply:
It arent't that many more. In most cases it's infect only one: like automations or using Siri. Even if you use the slider I hardly get more then two calls. If you use the step increment in Home Assistant, it's more.
Until a moment ago I just had a feeling not to use it since `timer` is imported from `threading` which usually means that it is somehow blocking something else. I looked it up and `timer` initialize a new thread with each init call. In Home Assistant we try to use async as much as possible to improve performance even on something like a Raspberry Pi. That isn't to say that the timer function couldn't be build using sth like `asyncio`, but IMO it is not worth it at the moment. There are bigger projects to work on, like support for Z-Wave device (which sadly will require some major changes to all `homekit` files) or making the `pyhap` libary asynchronous.